### PR TITLE
Added support for integer to complex conversion

### PR DESCRIPTION
**Pull request by BaneTrifa**

### CODE REVIEW

File: lib/Conversion/Utils/Utils.cpp [hunk 1]
```diff
@@ -379,14 +379,27 @@ Value convertScalarToDtype(OpBuilder &b, Location loc, Value scalar, Type dtype,
...
-        ;
       } else {
         realVal = scalar;
       }
 
       return b.create<complex::CreateOp>(loc, dtypeComplex, realVal, imgVal);
     }
 
+    // Int to complex type.
+    if (auto dtypeFloat = dyn_cast<mlir::IntegerType>(scalarType)) {
```
Comment (vivekkhandelwal1) [inline lib/Conversion/Utils/Utils.cpp:390]: `dtypeFloat` -> `dtypeInt`

File: lib/Conversion/Utils/Utils.cpp [hunk 2]
```diff
@@ -379,14 +379,27 @@ Value convertScalarToDtype(OpBuilder &b, Location loc, Value scalar, Type dtype,
...
+      auto complexElementType =
+          cast<mlir::FloatType>(dtypeComplex.getElementType());
+
+      Value realVal =
+          b.create<arith::SIToFPOp>(loc, complexElementType, scalar);
+      ;
```
Comment (vivekkhandelwal1) [inline lib/Conversion/Utils/Utils.cpp:396]: Please remove this.

**Review by vivekkhandelwal1 — COMMENTED** [review]
Please address the 2 small comments.

**Review by vivekkhandelwal1 — APPROVED** [review]

**Comment by BaneTrifa** [comment]
Can you merge this?

**Pull request merged**